strawberryfields/ops.py

```diff
@@ -1806,7 +1818,18 @@ class S2gate(Gate):
     """
     ns = 2
 
-    def __init__(self, r, phi=0.0):
+    def __init__(self, r, phi=None):
+        if (np.iscomplex([r, phi])).any():
+            raise ValueError("The arguments of S2gate(r, phi) cannot be complex")
+        if phi is None:
+            phi = 0.0
+            # TODO: remove warning in the new release
+            warnings.warn(
+                f"""Warning: since strawberryfields version v0.15.0,
+            Sgate(r, phi) takes two real arguments which represent
+            the polar decomposition of the complex squeezing parameter.
+            Using the arguments r={r} and phi=0.0"""
+            )
```

Comment:
Link for the S2gate:
https://github.com/XanaduAI/strawberryfields/blob/3c7b2a7bec0e05cf2133643bc4f63643db2e5a0e/strawberryfields/ops.py#L1813

Reply:
I don't think this warning is needed. Only an error must be raised if r is complex (not sure if this is already done).